Changed client structure

jBenderToledo opened the pull request:
Untested. Intent is to make each message check hidden by the API for maintainability reasons.

jBenderToledo committed:
Changed client structure

Untested. Intent is to make each message check hidden by the API for maintainability reasons.

jBenderToledo committed:
Update bot.py

Forgot to remove a ctx in making the code more legible.

joey-kilgore commented on bot.py:
```diff
@@ -8,32 +8,48 @@
-client = discord.Client()
+client = commands.Bot(command_prefix = '!')
+
+#####################
```
I think we need to discuss our coding style because I'm not sure how much we will want these big block comments. Especially because every method has the ```@client.event``` or ```@client.command``` before it.

jBenderToledo replied:
Yes, we should. What sort of standard should we go by? I figured that it would make it easier to read -- sorry if that was a bit silly.

joey-kilgore committed:
First update from coding 8/2/2019

Upload and download work. Logger started, and terminal commands started.

Co-Authored-By: Jacob Hunt <jacobhuntemail@gmail.com>
Co-Authored-By: Jared Maki <jared-maki@users.noreply.github.com>
Co-Authored-By: Jonathan Bender <jonathon.bender@rockets.utoledo.edu>

joey-kilgore committed:
added in the new logger, and replaced the print statements in the bot to now go through the logger

jBenderToledo committed:
Added a ping command and added app.log to .gitignore.

Jacob-Hunt committed:
No more "type error" exception

The test code for "terminal.py" now runs without crashing.  The output seems to get cut off after a certain string length is reached, which will not work for simulating a terminal environment: even just listing the files in the current directory results in a return string which gets cut off.  In the docs for the "Popen.communicate()" method, there is a note which says "The data read is buffered in memory, so do not use this method if the data size is large or unlimited" (https://docs.python.org/3/library/subprocess.html#subprocess.Popen).  So it seems that the issue is likely the use of the "Popen.communicate()" function when a more appropriate function should be used instead.

jBenderToledo committed:
Made a more standardized method of logging commands and operands.
Everything concerning starting a command should no longer be rewritten at the start of each new command.

jBenderToledo committed:
Merge branch 'improve-bot-structure' of https://github.com/cse-uprc/Bash-Bot into improve-bot-structure

jBenderToledo committed:
Modified the logger so that it prints in console the same way that it would print into the log, exceptions aside.

jBenderToledo committed:
Added header comments to each method currently in use in bot.py.
Added a log statement to the shutdown command that verifies that the bot has indeed logged out.

Jacob-Hunt committed:
Code Style and Documentation

Fixed typo, made variable names more descriptive, updated comments

Jacob-Hunt committed:
Added return dictionary to "executeCommand()"

Next issue to fix is replacing "self.terminalInstance.communicate()" (line 29) with a different function which won't cut off the return string after reaching a character limit. The fix will likely be found in the documentation for "subprocess.Popen" (see here: https://docs.python.org/3.7/library/subprocess.html#subprocess.Popen).

joey-kilgore committed:
added stuff for encoding and decoding text with communicate. Also added test code for running on linux.

Jacob-Hunt committed:
Edits from Live Share collaboration session

Co-Authored-By: Ben Doverson <thetoilettyrant@users.noreply.github.com>
Co-Authored-By: Jared Maki <jared-maki@users.noreply.github.com>
Co-Authored-By: Jacob Hunt <jacobhuntemail@gmail.com>

joey-kilgore committed:
Terminal commands can now be sent with !bash and the directory can be changed with !cwd

joey-kilgore commented on terminal.py:
```diff
@@ -0,0 +1,57 @@
+import os
+from pathlib import Path
+
+# Class to represent an instance of a terminal which is initiated
```
In bot.py we have the method/class headers on the lines following the method/class, but here they are before. We need to pick one.

Jacob-Hunt replied:
Added to to-do list for 8/23

joey-kilgore commented on terminal.py:
```diff
@@ -0,0 +1,57 @@
+            'errorString': stderr.decode('utf-8')
+        })
+
+# ~~~~~~~~~~~~~~~~ TEST CODE (for windows) ~~~~~~~~~~~~~~~~
```
The current state has not been tested for windows, can we test the changing directory with the optional currentDirectory parameter. @Jacob-Hunt

Jacob-Hunt replied:
@joey-kilgore Can you rephrase this comment? What, specifically, do you want tested for windows? Thanks!

joey-kilgore replied:
There is an optional third parameter for setting the current directory when creating a terminal instance. We should test this and ensure you can have different terminal directories called when instantiating bother the Terminal class and the Popen class

Jacob-Hunt replied:
@joey-kilgore Got it! Will test this on my Windows machine and report back shortly...

Jacob-Hunt replied:
@joey-kilgore Looks like we're good:

![terminal-py_execution](https://user-images.githubusercontent.com/26048894/63218442-f385de00-c128-11e9-827b-f80757eb02e7.PNG)

joey-kilgore commented on bot.py:
```diff
@@ -2,38 +2,213 @@
+
+
+	
+# The following segment of code was written by @TheToiletTyrant (316thebrains)
```
This code is irrelevant since we have git commits with authors. Unless we are programming before source control was created this isn't necessary

joey-kilgore commented on bot.py:
```diff
@@ -2,38 +2,213 @@
+# The following segment of code was written by @TheToiletTyrant (316thebrains)
+
+#############################################################
+# This code has been QUARENTINED due to it's infinite loops #
```
Should we remove this then for the first release, or should we try and fix it. It seems extremely different then the rest of the commands (using sockets to make a ping) and maybe should be part of a later release with some sort of web scraping functionality?

Jacob-Hunt replied:
Maybe clear it for the first release and create a "suggested features" issue-page for it?

joey-kilgore reviewed (commented):
Overall the functionality is there, but this needs to be cleaned up before the first release.

Jacob-Hunt committed:
Made header comments consistent across files

Jacob-Hunt committed:
Moved "!ping" from comment block to feature-request page

Jacob-Hunt committed:
Moved terminal test scripts to separate files. Began work on code style revision.

Jacob-Hunt committed:
Code-style edits/revision

joey-kilgore committed:
Moved the initialization of the client up to the beginning of bot.py
Also added ``` to format output and error messages for the bash commands

joey-kilgore committed:
Delete ProjectSettings.json

Remove random json file presumably generated from visual studio

joey-kilgore committed:
Delete VSWorkspaceState.json

Remove random file presumably generated from visual studio

joey-kilgore reviewed (commented):
@Jacob-Hunt if you delete the .suo and .sqlite files (presumably generated when someone opened this in visual studo) we can finish this.

Jacob-Hunt committed:
Delete slnx.sqlite

Jacob-Hunt committed:
Delete .suo
